sotodlib/mapmaking/ml_mapmaker.py

```diff
@@ -346,3 +425,38 @@ def write(self, prefix, tag, m):
         with h5py.File(oname, "w") as hfile:
             hfile["data"] = m
         return oname
+
+    def _checkcompat(self, other):
+        if other.cut_type != self.cut_type:
+            raise ValueError("Cut type mismatch")
+
+    def translate(self, other, junk):
+        """Translate junk degrees of freedom from one SignalCut representation
+        to another, e.g. changing sample rate. Returns the result"""
+        self._checkcompat(other)
+        res = np.full(self.off, -1e10, self.dtype)
+        for id in self.data:
+            sdata = self .data[id]
+            odata = other.data[id]
+            so3g.translate_cuts(odata.pcut.cuts, sdata.pcut.cuts, sdata.pcut.model, sdata.pcut.params, junk[odata.i1:odata.i2], res[sdata.i1:sdata.i2])
```

Comment:
@amaurea `so3g` is not imported.

Reply:
I see. `star` imports confuse linters and I thought it was not imported. If it works, it's okay.